ts/Grid/Core/Table/Actions/ColumnSorting.ts

```diff
@@ -142,6 +142,32 @@ class ColumnSorting {
         }
     }
 
+    /**
+     * Updates the column options with the new sorting state.
+     *
+     * @param col
+     * The column to update.
+     */
+    private updateColumnOptions(col: Column): void {
+        const order = col.viewport.grid.querying.sorting.currentSorting?.order;
+
+        if (col.id === this.column.id && order) {
+            col.update({
+                sorting: {
+                    order
+                }
+            }, false);
+        } else {
+            delete col.options.sorting?.order;
```

Comment:
Why do we need to delete it here? In the next lines, you check sorting, and then delete it.

Reply:
First, we remove the order of all previously set columns, and second, we clear the sorting option object if there are no other options there.